### PR TITLE
fix(json-magic): resolve external references on windows

### LINKED ISSUE #8131: json-magic fails with EXTERNAL_REFERENCE_NOT_FOUND on windows
### What happens?

Given the below specs running
```ts
const spec = await bundle(<path to api.yaml>, {
    plugins: [
      readFiles(),
      fetchUrls(),
      parseJson(),
      parseYaml(),
    ],
    treeShake: false,
  });
const { valid, errors } = await validateSpec(dereferencedData);
```
yields a `spec` that's missing the `x-ext` key and `errors` have
```js
{
  code: "EXTERNAL_REFERENCE_NOT_FOUND",
  message: "Can't resolve external reference: ./schemas/pet.yaml",
}
```
strangely this seems to work on linux. You can see the CI runs [here](https://github.com/orval-labs/orval/pull/2944)

### Given spec
```yml
# api.yaml
openapi: '3.0.0'
info:
  version: 1.0.0
  title: Swagger Petstore
  license:
    name: MIT
servers:
  - url: http://petstore.swagger.io/v1
paths:
  /my-path:
    put:
      tags:
        - putPet
      operationId: putPet
      requestBody:
        content:
          'application/json':
            schema:
              $ref: '#/components/schemas/Pet'
        required: true
      responses:
        '200':
          description: OK
          content:
            'application/json':
              schema:
                $ref: '#/components/schemas/Pet'
components:
  schemas:
    Pet:
      $ref: 'schemas/pet.yaml#/components/schemas/Pet'
```

```yml
# schemas/pet.yaml
openapi: 3.0.2
info:
  title: PetStore
  version: 1.0.0
paths: {}
components:
  schemas:
    Pet:
      type: object
      properties:
        id:
          type: string
        name:
          type: string
        owner:
          $ref: './person.yaml#/components/schemas/Person'
```
```yml
# schemas/person.yaml
openapi: 3.0.2
info:
  title: PetStore
  version: 1.0.0
paths: {}
components:
  schemas:
    Person:
      type: object
      required:
        - id
        - name
      properties:
        id:
          type: string
        name:
          type: string
```

### What did you expect to happen?

I was expecting `spec` to have the `x-ext` key with the external specs.

### OpenAPI Document

_No response_

### PR DESCRIPTION
**Pull request by DemonHa**
Fixes #8131

We were relying on a custom path polyfill that did not correctly handle Windows-style paths, leading to inconsistent behavior across platforms.

This change replaces the polyfill with pathe, which is browser-friendly and correctly supports both Windows and POSIX file paths, ensuring consistent path resolution in all environments.

## Checklist

- [x] I explained why the change is needed.
- [x] I added a changeset. 
- [x] I added tests.
- [ ] I updated the documentation.

---

> [!NOTE]
> **Medium Risk**
> Touches core path-resolution logic used for bundling/dereferencing external documents, so regressions could break reference loading across environments; mitigated by extensive new unit tests including Windows cases.
> 
> **Overview**
> Fixes cross-platform external reference resolution by replacing the custom `path` polyfill with `pathe` and ensuring URL path resolution/relativization uses `path.posix` (avoiding Windows separator/drive-letter quirks in `resolveReferencePath`/`toRelativePath`).
> 
> Adds broad new test coverage for absolute/relative/local/URL and Windows-style path edge cases, and updates `workspace-store` filesystem-loading tests to expect CWD-resolved reference paths. Includes a changeset to ship patch releases for `@scalar/json-magic` and `@scalar/workspace-store`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 864d94c252dd6c672c425e28a4a2e8de200957fd. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: packages/json-magic/package.json [hunk 1]
```diff
@@ -84,6 +84,7 @@
   ],
   "dependencies": {
     "@scalar/helpers": "workspace:*",
+    "pathe": "^2.0.3",
```
Comment (hanspagel) [inline packages/json-magic/package.json:87]: adds 6 kb, has 55M downloads/week, seems legit

**Review by hanspagel — APPROVED** [review]
cool
we love windows
